include/soci/soci-platform.h

```diff
@@ -8,6 +8,26 @@
 #ifndef SOCI_PLATFORM_H_INCLUDED
 #define SOCI_PLATFORM_H_INCLUDED
 
+// Check some C++11 Features
+#if     defined(_MSC_VER)
+// Bug in Visual Studio: __cplusplus still means C++03
+// https://connect.microsoft.com/VisualStudio/feedback/details/763051/
+#define SOCI_HAVE_NOEXCEPT (_MSC_VER >= 1900)
+#define SOCI_HAVE_LAMBDA   (_MSC_VER >= 1600)
+#elif   defined(__clang__) || defined(__GNUG__)
```

Comment:
On 8 May 2015 at 11:16, tvb377 <notifications@github.com> wrote:

> Line 65 in SociConfig.cmake throws a FATAL_ERROR when using a compiler
> other than clang/gcc/vc.
>

This is SociConfig.cmake caveat which should be fixed at some point.
(Slowly, I try to remove as much as possible of our *.cmake customisations
and rely on CMake built-ins).


> Neither do I know if __cplusplus is defined correctly for compilers other
> than MSVC,gcc and clang nor do I know if CMake reports this compiler as
> being something else nor if all other compilers have implemented noexcept.
>

We don't support compilers that don't define or define incorrectly the
standard predefined macro __cplusplus. So, we can safely rely on it.


> Unless you know these compilers well and you are about to remove this line
> 65 in SociConfig.cmake I'd keep the change like I proposed and leave the
> necessary changes to a user of other compilers.
>

I've removed the FATAL_ERROR restriction, see
1d26033e060c2108d4ffbe06bbe4a6895c24bdc7.
It was too restrictive indeed. Actually, it was more like a left-over from
early days of CMake configuration development.
Thanks for pointing that.